src/librustdoc/html/render.rs

```diff
@@ -143,6 +144,33 @@ impl Impl {
     }
 }
 
+#[derive(Debug)]
+pub struct IoError {
+    filename: String,
+    error: io::Error,
+}
+
+impl Error for IoError {
+    fn description(&self) -> &str {
+        self.error.description()
+    }
+}
+
+impl Display for IoError {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "\"{}\": {}", self.filename, self.error)
+    }
+}
+
+impl From<io::Error> for IoError {
```

Comment:
How many locations does this end up getting used? If it's enough that this can't be removed then the `Display` implementation is probably going to want to change to account for the empty filename.